sql/analyzer/aliases.go

```diff
@@ -286,6 +288,14 @@ func renameAliases(node sql.Node, oldNameLower string, newName string) (sql.Node
 			}
 		}
 
+		// if the node is a table and its parent is not a table alias, add a table alias node
+		if _, ok := newNode.(sql.Table); ok {
```

Comment:
This might be missing subquery aliases and table functions, maybe others. And nit: the plan looks a bit nicer with this, but practically speaking `TableAlias(A)->TableAlias(B)->Tablescan(C)` is equivalent to `TableAlias(A)->Tablescan(C)`. `transform.NodeWithCtx` is slightly more expensive than `transform.Node`, and the logic can be harder to follow when parent checking isn't strictly necessary. Other rules duplicate table aliases in a maybe undesirable way; having a `deduplicateTableAliases` rule that does: `(TableAlias1->TableAlias2)=>(TableAlias1)` might be useful at some point.

Something like this would maybe catch the other nameable nodes:

```golang
switch node := node {
case sql.NameableNode:
  newNode = plan.NewTableAlias(newName, node)
default:
}
```

Reply:
Reverted this change.

Reply:
PR updated. We're now directly renaming TableAlias, SubqueryAlias, and Procedure. And in case where we see a Nameable with the old name, we rename by introducing a TableAlias on top of the node.